connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -432,7 +439,9 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
 
             final SchemaAndValue value;
             try {
-                value = converter.toConnectData(topic, record.value());
+                synchronized (lock) {
```

Comment:
Yeah, I'm not sure we need to provide the synchronization either. Serializers are currently assumed to be synchronized by the producer, for example. So there is some precedent for that assumption, although it only needs to be thread safe if you use it from multiple threads. Currently Deserializer doesn't need to be since the consumer is single threaded anyway.

@kkonstantine Do we know of a case of one that isn't thread safe already? iirc JsonConverter uses synchronized caches which is the only state I can think of that it has.